src/Sulu/Bundle/AdminBundle/Resources/js/components/Select/Select.js

```diff
@@ -118,6 +189,70 @@ class Select<T> extends React.Component<Props<T>> {
 
     handleOptionListClose = this.closeOptionList;
 
+    @action requestFocus = (elementIndex: number) => {
+        if (!this.buttonRefsByIndex.has(elementIndex)) {
+            return;
+        }
+
+        this.focusedElementIndex = elementIndex;
+        const ref = this.buttonRefsByIndex.get(elementIndex);
+
+        if (ref) {
+            ref.focus();
+        }
+    };
+
+    handleRequestFocus = (elementIndex: number) => () => {
+        this.requestFocus(elementIndex);
+    };
+
+    @action handleKeyDown = (event: KeyboardEvent) => {
+        if (!this.open) {
+            return;
+        }
+
+        if (event.key === 'Escape') {
+            event.preventDefault();
+            this.closeOptionList();
+            this.clearSearchText();
+
+            return;
+        }
+
+        let focusedElementIndex = this.focusedElementIndex;
+
+        if (event.key === 'ArrowUp') {
+            event.preventDefault();
+            this.clearSearchText();
+
+            focusedElementIndex = Math.max(
+                ...this.availableButtonIndices.filter((i) => i < this.focusedElementIndex)
+            );
+        }
+
+        if (event.key === 'ArrowDown') {
+            event.preventDefault();
+            this.clearSearchText();
+
+            focusedElementIndex = Math.min(
```

Comment:
if we remove the `this.focusedElementIndex`, we could just directly call `requestFocus`

Reply:
no, we need the focusedElementIndex, otherwise we can't know which one is the previous/next element

Reply:
yes, i forgot about that. but we could still just call `requestFocus` here directly without the need of a variable 🙂

Reply:
fixed